### ISSUE TITLE
Additional Fields not working 

### ISSUE THREAD

**Issue by ghost** [sorry-app/devise_campaignable#14]
Im sending additional fields to Mailchimp but they are not being created.

Has Mailchimp's API change? Any suggestions on what to try?

config.campaignable_additional_fields 

`
config.campaignable_additional_fields = [:email, :name, :stripe_subscription_id, :stripe_id, :created_at]`

**Comment by ghost** [comment]
Okay - so I tried adding the following fields on my mailing list. `Subscription_id` and `Stripe_id`. But I need to find a way to match them with the campaignable additional fields. 

I also tried adding a Default Merge Tag Values on Mailchimps side.

**Comment by SirRawlins** [comment]
Hi Lauro,

I don't believe that the MailChimp API has changed. Looking at the code and
Gibbon docs I can't see why this isn't working.

I'm not in the office at the moment but can take a closer look first thing
tomorrow for you.

Robert

**Comment by ghost** [comment]
Hi Robert, 

Thanks for the response. Let me know if you need any more info to check this out. 

Here's my User Model:

```ruby
  devise :database_authenticatable, :registerable,
         :recoverable, :rememberable, :trackable, :validatable,
         :omniauthable, :campaignable
```

My devise initializer: 

```ruby
Devise.setup do |config|

  config.campaignable_api_key = 'xxxxxx'
  config.campaignable_list_id = 'xxxxxxx'
  config.campaignable_additional_fields = [:email, :name, :stripe_subscription_id, :stripe_id]

end
```

Also the callback: 

```
 => {"id"=>"d695463ac7", "status"=>"pending", "total_operations"=>0, "finished_operations"=>0, "errored_operations"=>0, "submitted_at"=>"2017-04-25T17:43:38+00:00", "completed_at"=>"", "response_body_url"=>"", "_links"=>[{"rel"=>"parent", "href"=>"https://us10.api.mailchimp.com/3.0/batches", "method"=>"GET", "targetSchema"=>"https://us10.api.mailchimp.com/schema/3.0/Definitions/Batches/CollectionResponse.json", "schema"=>"https://us10.api.mailchimp.com/schema/3.0/CollectionLinks/Batches.json"}, {"rel"=>"self", "href"=>"https://us10.api.mailchimp.com/3.0/batches/d695463ac7", "method"=>"GET", "targetSchema"=>"https://us10.api.mailchimp.com/schema/3.0/Definitions/Batches/Response.json"}, {"rel"=>"delete", "href"=>"https://us10.api.mailchimp.com/3.0/batches/d695463ac7", "method"=>"DELETE"}]} 
```


Appreciate the help!

**Comment by SirRawlins** [comment]
Thanks for the extra info. Looking at my setup here i think my custom
fields are missing too, so looks like we may have a bug.

I will let you know what I find tomorrow.

**Comment by SirRawlins** [comment]
@lauroengineer 

**I've just released 0.2.3 which includes a fix for this issue**.

Seems that after the upgrade from MailChimp API v2 -> v3 we needed to submit the merge_fields in uppercase, so the plugin now up-cases them before sending.

The other thing to be sure of is that you have merge_fields configured on the list in MailChimp which match your attributes.

**So if your attributes look like this:**

```
config.campaignable_additional_fields = [:name, :stripe_subscription_id, :stripe_id]
```

**In your MailChimp account, you should have merge_fields configured like this:** 

```
NAME, STRIPE_SUBSCRIPTION_ID and STRIPE_ID.
```

Give this a go and let me know how you get on.

**Comment by ghost** [comment]
Sorry about the late reply. Was on vacation. 

Thanks for the quick fix. It worked perfectly. 

Thanks @SirRawlins!
